plutus-ledger-constraints/src/Ledger/Constraints/TxConstraints.hs

```diff
@@ -298,7 +308,6 @@ mustIncludeDatum = singleton . MustIncludeDatum
 mustPayToTheScript :: forall i o. PlutusTx.ToData o => o -> Value -> TxConstraints i o
 mustPayToTheScript dt vl =
     mempty { txOwnOutputs = [ScriptOutputConstraint dt vl] }
-    <> mustIncludeDatum (Datum (PlutusTx.toBuiltinData dt))
```

Comment:
@sjoerdvisscher Why was this removed?

Reply:
`ScriptOutputConstraint` already does with the datum what `mustIncludeDatum` does, both on-chain and off-chain. And for ledger-tx-constraints it doesn't make sense to split it up like this. Same for the other cases.